internal/fromproto5/arguments_data_test.go

```diff
@@ -717,3 +720,2111 @@ func TestArgumentsData(t *testing.T) {
+			input: []*tfprotov5.DynamicValue{
+				DynamicValueMust(tftypes.NewValue(tftypes.Bool, true)),
+				DynamicValueMust(tftypes.NewValue(tftypes.String, "true")),
+				DynamicValueMust(tftypes.NewValue(tftypes.String, "false")),
+			},
+			definition: function.Definition{
+				Parameters: []function.Parameter{
+					function.BoolParameter{
+						Validators: []function.BoolParameterValidator{
+							testvalidator.Bool{
+								ValidateMethod: func(ctx context.Context, req function.BoolParameterValidatorRequest, resp *function.BoolParameterValidatorResponse) {
+									got := req.Value
+									expected := types.BoolValue(true)
+
+									if !got.Equal(expected) {
+										resp.Error = function.NewArgumentFuncError(
+											req.ArgumentPosition,
+											"Error Diagnostic: This is an error.",
+										)
+									}
+								},
+							},
+						},
+					},
+				},
+				VariadicParameter: function.StringParameter{
+					Validators: []function.StringParameterValidator{
+						testvalidator.String{
+							ValidateMethod: func(ctx context.Context, req function.StringParameterValidatorRequest, resp *function.StringParameterValidatorResponse) {
+								got := req.Value
+								expected := types.StringValue("true")
+
+								if !got.Equal(expected) {
+									resp.Error = function.NewArgumentFuncError(
+										req.ArgumentPosition,
+										"Error Diagnostic: string validator error.",
+									)
+								}
+							},
+						},
+					},
+				},
+			},
+			expected: function.NewArgumentsData([]attr.Value{
+				basetypes.NewBoolValue(true),
+			}),
+			expectedFuncError: function.ConcatFuncErrors(
+				function.NewArgumentFuncError(2, "Error Diagnostic: string validator error."),
+			),
+		},
+	}
+
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got, diags := fromproto5.ArgumentsData(context.Background(), testCase.input, testCase.definition)
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+
+			if diff := cmp.Diff(diags, testCase.expectedFuncError); diff != "" {
+				t.Errorf("unexpected diagnostics difference: %s", diff)
+			}
+		})
+	}
+}
+
+func createListValue(elementType attr.Type, elements []attr.Value) basetypes.ListValue {
+	list, _ := basetypes.NewListValue(elementType, elements)
+	return list
+}
+
+func createMapValue(elementType attr.Type, elements map[string]attr.Value) basetypes.MapValue {
+	mapVal, _ := basetypes.NewMapValue(elementType, elements)
+	return mapVal
+}
+
+func createObjectValue(attributeTypes map[string]attr.Type, attributes map[string]attr.Value) basetypes.ObjectValue {
+	object, _ := basetypes.NewObjectValue(attributeTypes, attributes)
+	return object
+}
+
+func createSetValue(elementType attr.Type, elements []attr.Value) basetypes.SetValue {
+	list, _ := basetypes.NewSetValue(elementType, elements)
+	return list
+}
```

Comment:
Can we use the `New{Type}ValueMust` equivalents in-place of these? (similar comment for the other `fromproto6` test)

Reply:
Thanks for pointing this out, I didn't know about the `New{Type}ValueMust` functions.